### ISSUE TITLE
Error E0277 does not detected for `str` field in struct

### ISSUE THREAD

**Issue by neonaot** [intellij-rust/intellij-rust#8722]
## Environment

* **IntelliJ Rust plugin version:** 0.4.169.4542-221-nightly
* **Rust toolchain version:** 1.61.0-nightly (f0c4da499 2022-03-01) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.1 Beta (CL-221.5080.93)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
<img width="774" alt="image" src="https://user-images.githubusercontent.com/49211026/161938274-47563b48-b0e6-42f1-a760-218729a6a15c.png">
error[E0277]: the size for values of type `str` cannot be known at compilation time

## Steps to reproduce
```rust
fn main() {}

// error
struct S{
  s: str,
  i: u8
}

// no error by compiler
struct S2 {
    s: str
}

```
